.github/workflows/ci.yml

```diff
@@ -77,25 +125,108 @@ jobs:
+            git commit -m "Update challenge scores"
+            git push origin HEAD:${{ github.ref }}
+          else
+            echo "The challenge scores didn't change."
+          fi
+
+      - id: push_cassettes
+        name: Push updated cassettes
+        run: |
+          if [[ "${{ startsWith(github.event_name, 'pull_request') }}" = "true" ]]; then
+            is_pull_request=true
+            cassette_branch="${{ github.event.pull_request.user.login }}-${{ github.event.pull_request.head.ref }}"
+            base_branch="${{ github.event.pull_request.base.ref }}"
+          else
+            current_branch=$(echo ${{ github.ref }} | sed -e "s/refs\/heads\///g")
+            cassette_branch=$current_branch
+          fi
+
+          cd tests/Auto-GPT-test-cassettes
+          git fetch origin $cassette_branch
+
+          # Commit & push changes to cassettes if any
+          if ! git diff-index --quiet $cassette_branch; then
+            git add .
+            git commit -m "Auto-update cassettes"
+            git pull --rebase origin $cassette_branch
+
+            git push origin HEAD:$cassette_branch
+
+            cd ../..
+            if [ $is_pull_request ]; then
+              git fetch origin $base_branch
+              cassette_diff=$(git diff $cassette_branch origin/$base_branch)
+            else
+              git add tests/Auto-GPT-test-cassettes
+              git commit -m "Update cassette submodule"
+              git push origin HEAD:$current_branch
+            fi
+          else
+            echo "No cassette changes to commit"
+          fi
+
+          if [ -n "$cassette_diff" ]; then
+            echo "updated=true" >> $GITHUB_OUTPUT
+          else
+            echo "updated=false" >> $GITHUB_OUTPUT
+          fi
+
+      - name: Post Set up git token auth
+        run: |
+          git config --unset "${{ steps.setup_git_auth.outputs.config_key }}"
+
+      - name: Apply or remove behaviour change label and comment on PR
+        if: ${{ startsWith(github.event_name, 'pull_request') }}
         run: |
-          git add tests
-          git diff --cached --quiet && echo "No changes to commit" && exit 0
-          git config user.email "github-actions@github.com"
-          git config user.name "GitHub Actions"
-          git commit -m "Add new cassettes"
-          git checkout -b cassette-diff-${{ github.event.pull_request.number }}
-          git remote add target https://${{ secrets.GITHUB_TOKEN }}@github.com/${{ github.event.pull_request.base.repo.full_name }}
-          git push -f target cassette-diff-${{ github.event.pull_request.number }}
+          PR_NUMBER=${{ github.event.pull_request.number }}
+          TOKEN=${{ secrets.PAT_REVIEW }}
+          REPO=${{ github.repository }}
+
+          if [[ "${{ steps.push_cassettes.outputs.updated }}" == "true" ]]; then
+            echo "Adding label and comment..."
+            curl -X POST \
+            -H "Authorization: Bearer $TOKEN" \
```

Comment:
Don’t save token to a var? Just use it directly? Also does this log?

Reply:
It's scrubbed from the log because it's a secret